asset-transfer-abac/README.md

```diff
@@ -0,0 +1,176 @@
+```
+
+## Transfer the asset
+
+As the owner of Asset1, the creator1 identity has the ability to transfer the asset to another owner. In order to transfer the asset, the owner needs to provide the Base64 encoded name and issuer of the new owner to the `TransferAsset` function. The `asset-transfer-abac` smart contract has a `GetEncodedID` function that allows users to retrieve their Base64 encoded certificate information and provide it to the asset owner out of band (we omit this step). Issue the command below to transfer Asset1 to an identity that was created by the test network script:
+```
+peer chaincode invoke $TARGET_TLS_OPTIONS -C mychannel -n abac -c '{"function":"TransferAsset","Args":["Asset1","eDUwOTo6Q049dXNlcjEsT1U9Y2xpZW50LE89SHlwZXJsZWRnZXIsU1Q9Tm9ydGggQ2Fyb2xpbmEsQz1VUzo6Q049Y2Eub3JnMS5leGFtcGxlLmNvbSxPPW9yZzEuZXhhbXBsZS5jb20sTD1EdXJoYW0sU1Q9Tm9ydGggQ2Fyb2xpbmEsQz1VUw=="]}'
+```
+Query the ledger to verify that the asset has a new owner:
+```
+peer chaincode query -C mychannel -n abac -c '{"function":"ReadAsset","Args":["Asset1"]}'
+```
+We can see that Asset1 with is now owned by User1. The certificate information is decoded by the smart contract.
+```
+{"ID":"Asset1","color":"blue","size":20,"owner":"x509::CN=creator1,OU=client+OU=org1,O=Hyperledger,ST=North Carolina,C=US::CN=ca.org1.example.com,O=org1.example.com,L=Durham,ST=North Carolina,C=US","appraisedValue":100}
+```
+
+## Update the asset
+
+Now that the asset has been transferred, the new owner can update the asset properties. The smart contract uses the `GetID()` API to ensure that the update is being submitted by the asset owner. Run the following command to set the MSP path to User1:
+```
+export CORE_PEER_MSPCONFIGPATH=${PWD}/organizations/peerOrganizations/org1.example.com/users/User1@org1.example.com/msp
+```
+
+We can now update the asset. Run the following command to change the asset color from blue to green. All other aspects of the asset will remain unchanged.
+```
+peer chaincode invoke $TARGET_TLS_OPTIONS -C mychannel -n abac -c '{"function":"UpdateAsset","Args":["Asset1","green","20","100"]}'
+```
+Run the query command again to verify that the asset has changed color:
+```
+peer chaincode query -C mychannel -n abac -c '{"function":"ReadAsset","Args":["Asset1"]}'
+```
+The result will display that Asset1 is now green:
+```
+{"ID":"Asset1","color":"green","size":20,"owner":"x509::CN=creator1,OU=client+OU=org1,O=Hyperledger,ST=North Carolina,C=US::CN=ca.org1.example.com,O=org1.example.com,L=Durham,ST=North Carolina,C=US","appraisedValue":100}
+```
+
+## Delete the asset
+
+While the owner can update or transfer the asset, the asset can only be deleted by an identity with the `abac.creator=true` attribute. Try to delete the asset using the User1 identity without the attribute:
```

Comment:
As I said in the code comments, it is a little awkward to delete based on 'creator' role. I suggest to check ownership instead.

Reply:
done.